components/common/src/types.rs

```diff
@@ -241,6 +245,15 @@ impl fmt::Debug for EventStreamServerCertificate {
     }
 }
 
+impl PartialEq<EventStreamServerCertificate> for EventStreamServerCertificate {
+    fn eq(&self, other: &EventStreamServerCertificate) -> bool {
+        match (self.0.to_der(), other.0.to_der()) {
```

Comment:
Maybe because `to_der` is a failable operation which makes the logic a little awkward. Should two errors be equal? Since we only use it for testing I think this implementation is ok.

Reply:
Could you add a note that it's only for testing?

I don't suppose we can flag this with a `#[test]` attribute because of the crate it's in , right?

Reply:
Yes, that would be good.

Right :disappointed:. At some point I think it would be interesting to do some high level refactorings on what types/functions are contained in what crates.